crates/router/src/core/payments.rs

```diff
@@ -788,6 +788,7 @@ pub async fn payments_core<F, Res, Req, Op, FData>(
     state: SessionState,
     req_state: ReqState,
     merchant_account: domain::MerchantAccount,
+    _profile_id_list: Option<Vec<String>>,
```

Comment:
Why is this `Option<Vec<String>>`?

Reply:
This would give a bit more flexibility in future.

Some core functions are being called from other flows like webhooks, compatibility layer etc. So in these parts it wouldn't make sense to pass hard-coded empty vecs.